### PR TITLE
Resolves uncaught "call is cancelled" subscription exception issue #202

### PR DESCRIPTION
**Pull request by TrekSoft**
*Issue #, if available:*
[Issue #202]

*Description of changes:*
Fixes bad runtime exceptions inside a thread with a callback onError wrapping the exception.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-android-sdk-appsync-runtime/src/main/java/com/apollographql/apollo/internal/RealAppSyncSubscriptionCall.java [hunk 1]
```diff
@@ -68,23 +68,22 @@ public void run() {
...
-                        case CANCELED:
-                            throw new RuntimeException("Cancelled", new ApolloCanceledException("Call is cancelled."));
+                    case CANCELED:
+                        callback.onFailure(new ApolloCanceledException("Call is cancelled."));
+                        break;
 
-                        case ACTIVE:
-                            throw new IllegalStateException("Already Executed");
+                    case ACTIVE:
+                        callback.onFailure(new ApolloException("Already Executed."));
```
Comment (kvasukib) [inline aws-android-sdk-appsync-runtime/src/main/java/com/apollographql/apollo/internal/RealAppSyncSubscriptionCall.java]: Minor. Let's remove the trailing period '.' to match with the existing message exactly.

File: aws-android-sdk-appsync-runtime/src/main/java/com/apollographql/apollo/internal/RealAppSyncSubscriptionCall.java [hunk 2]
```diff
@@ -68,23 +68,22 @@ public void run() {
...
+                        break;
 
-                        default:
-                            throw new IllegalStateException("Unknown state");
-                    }
+                    default:
+                        callback.onFailure(new ApolloException("Unknown State."));
```
Comment (kvasukib) [inline aws-android-sdk-appsync-runtime/src/main/java/com/apollographql/apollo/internal/RealAppSyncSubscriptionCall.java]: Minor. Let's remove the trailing period '.' to match with the existing message exactly.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,11 @@
 # Change Log - AWS AppSync SDK for Android
 
+## [Release 2.10.1](https://github.com/awslabs/aws-mobile-appsync-sdk-android/releases/tag/release_v2.10.1)
+
+### Bug Fixes
+* Fixed Uncaught "Call is cancelled" subscription exception See [issue#202](https://github.com/awslabs/aws-mobile-appsync-sdk-android/issues/202)
```
Comment (kvasukib) [inline CHANGELOG.md]: can you add more details like:

The callback passed in will be invoked when the subscriptions are cancelled, failed and in any other scenarios.

**Comment by TrekSoft** [comment]
All comments resolved and rebased with your CHANGELOG PR which I approved pushed

**Pull request merged**